packages/react-test-renderer/src/ReactShallowRenderer.js

```diff
@@ -99,6 +118,259 @@ class ReactShallowRenderer {
     this._rendering = false;
     this._forcedUpdate = false;
     this._updater = new Updater(this);
+    if (enableHooks) {
+      this._dispatcher = ReactCurrentDispatcher.current = this._createDispatcher();
+      this._workInProgressHook = null;
+      this._firstWorkInProgressHook = null;
+      this._isReRender = false;
+      this._didScheduleRenderPhaseUpdate = false;
+      this._renderPhaseUpdates = null;
+      this._currentlyRenderingComponent = null;
+      this._numberOfReRenders = 0;
+    }
+  }
+
+  _validateCurrentlyRenderingComponent() {
+    invariant(
+      this._currentlyRenderingComponent !== null,
+      'Hooks can only be called inside the body of a function component.',
+    );
+  }
+
+  _createDispatcher() {
```

Comment:
We'll want to add a return type of `typeof Dispatcher` to this one #14599 is merged